### PR TITLE
Optimised backend controllers and database queries

### PR DESCRIPTION
**Pull request by Saif807380**
- Used `.select()` and `.lean()`  in controllers and middleware to reduce database operations and response payload.
- Added new index on user schema
- Changed type of `authorID` in blog's schema to object ref

### CODE REVIEW

File: server/src/config/dbconnect.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ function connect() {
     useNewUrlParser: true,
     useUnifiedTopology: true,
     useFindAndModify: false,
+    autoIndex: false
```
Comment (ShubhankarKG) [inline server/src/config/dbconnect.js:10]: Is there a specific reason for this?

File: server/.gitignore [hunk 2]
```diff
@@ -1,4 +1,4 @@
 node_modules/
 npm-debug.log*
 .env
-uploads/
\ No newline at end of file
+uploads/
```
Comment (ShubhankarKG) [inline server/.gitignore:4]: This seems like a format diff to me.

File: server/src/config/dbconnect.js [hunk 3]
```diff
@@ -24,4 +25,4 @@ function connect() {
   mongoose.connection.on("error", console.error.bind(console, "Mongo Error"));
 }
 
-module.exports = connect;
\ No newline at end of file
+module.exports = connect;
```
Comment (ShubhankarKG) [inline server/src/config/dbconnect.js:28]: This seems like a format diff to me.

File: server/src/controllers/AuthController.js [hunk 4]
```diff
@@ -23,9 +23,9 @@ module.exports = {
 
       const user1 = await User.findOne({
         username: username,
-      });
+      }).lean().select({"username":1});
 
-      /* If user exists, return 422 - 
+      /* If user exists, return 422 -
```
Comment (ShubhankarKG) [inline server/src/controllers/AuthController.js:28]: This seems like a format diff to me.

File: server/src/controllers/Events.js [hunk 5]
```diff
@@ -147,7 +147,7 @@ module.exports = {
     try {
       const eventId = req.body.id;
       const userEmail = req.body.email;
-      const event = await Event.findById(eventId);
+      const event = await Event.findById(eventId).select({"eventName":1}).lean();
```
Comment (ShubhankarKG) [inline server/src/controllers/Events.js:150]: The immediate next line requires `event.date`, that would not be available here.

File: server/src/models/User.js [hunk 6]
```diff
@@ -47,6 +47,8 @@ const user = new mongoose.Schema({
     }
 })
 
+user.index({ username: 1 })
+
 const User = mongoose.model('users', user)
 
-module.exports = User;
\ No newline at end of file
+module.exports = User;
```
Comment (ShubhankarKG) [inline server/src/models/User.js:54]: This seems like a format diff to me.

**Review by ShubhankarKG — CHANGES_REQUESTED** [review]
Thanks, the changes look good. Just a few inline suggestions. It's okay to leave the format diff as is. :pray:

File: server/src/config/dbconnect.js (hunk 1 above)
Comment (Saif807380) [reply server/src/config/dbconnect.js:10]: That is for optimising the creation of indices during production. [Doc Reference](https://mongoosejs.com/docs/guide.html#indexes)

File: server/src/controllers/Events.js (hunk 5 above)
Comment (Saif807380) [reply server/src/controllers/Events.js:150]: Will fix that

File: server/src/config/dbconnect.js (hunk 1 above)
Comment (ShubhankarKG) [reply server/src/config/dbconnect.js:10]: Then its better if we keep it to 
`process.env.NODE_ENV !== "production"`

**Review by ShubhankarKG — APPROVED** [review]
Thanks, LGTM

**Pull request merged**